### DISCUSSION TITLE
french keyboard with ubuntu

### DISCUSSION THREAD

**Discussion by valgit** [utmapp/UTM#5579] [Linux]
**Describe the issue**  
Replace this text with a clear and concise description of what the issue is.

**Configuration**  
* UTM Version: 4.3.5 from github
* macOS Version: macos 15
* Mac Chip (Intel, M1, ...):  macbook Apple M1, french keyboard

when creating a new VM with Ubuntu Server ARM62 22.04, a french keyboard is detected, but after installation some keys are wrong and/or missing.
like : "=_@" ....

**Comment by adeadman** [comment]
You didn't say whether you are using Apple Virtualisation, but I suspect you are.
This sounds like a duplicate of #4041 
You can view the details in this writeup here: https://eclecticlight.co/2023/06/27/keyboard-layouts-in-lightweight-virtualisation/

**Comment by osy** [comment]
If it's AVM, see #4041, and if it's QEMU, you probably need to change your keyboard layout in Ubuntu settings.
